File: app/src/main/java/com/avdhootsolutions/aswack_shopkeeper/models/Categories.kt
package com.avdhootsolutions.aswack_shopkeeper.models

import com.google.gson.annotations.SerializedName

class Categories {
    @SerializedName("id")
    var id: String? = null

    @SerializedName("name")
    var name: String? = null

    @SerializedName("sub_text")
    var sub_text: String? = null

    @SerializedName("category_name")
    var category_name: String? = null

    @SerializedName("category")
    var category: String? = null

    @SerializedName("image")
    var image: String? = null

    var isChecked = false

}

File: lib/models/categories.dart
class Categories {
  String? id;
  String? name;
  String? subText;
  String? categoryName;
  String? category;
  String? image;
  bool isChecked = false;

  Categories.fromJson(Map<String, dynamic> json)
      : id = json['id']?.toString(),
        name = json['name']?.toString() ?? json['category_name']?.toString(),
        subText = json['sub_text']?.toString(),
        categoryName = json['category_name']?.toString(),
        category = json['category']?.toString(),
        image = json['image']?.toString();
}
